pennylane_lightning/src/tests/CMakeLists.txt

```diff
@@ -19,6 +19,9 @@ FetchContent_Declare(
 
 FetchContent_MakeAvailable(Catch2)
 
+get_target_property(CATCH2_IID Catch2 INTERFACE_INCLUDE_DIRECTORIES)
+set_target_properties(Catch2 PROPERTIES INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "${CATCH2_IID}")
```

Comment:
These lines ensure Catch2 code is excluded from `clang-tidy`.

Reply:
So none of the test macros get tidied? Or our Catch2 tests are excluded?

So the trick is to get the included headers from the library, and reclassify them as system includes, so then CT ignores them, as they are treated as not part of the package?

Reply:
We just want to exclude Catch2 header files from `clang-tidy` as `clang-tidy` does not like a few lines of the code here. The same technique is also used for Kokkos https://github.com/PennyLaneAI/pennylane-lightning/blob/736f7826772ca2182c7690c04af3454c72900037/cmake/process_options.cmake#L92-L95

Reply:
Got it, thanks.